ios/RNCImageEditor.mm

```diff
@@ -50,12 +50,14 @@ - (void) cropImage:(NSString *)uri
   CGSize size = [RCTConvert CGSize:@{ @"width": @(data.size().width()), @"height": @(data.size().height()) }];
   CGPoint offset = [RCTConvert CGPoint:@{ @"x": @(data.offset().x()), @"y": @(data.offset().y()) }];
   CGSize targetSize = size;
-  if (data.displaySize().has_value()) {
-    JS::NativeRNCImageEditor::SpecCropImageCropDataDisplaySize displaySize = *data.displaySize(); // Extract the value from the optional
+  CGSize displaySize = CGSizeZero;
+  BOOL hasDisplaySizeValue = data.displaySize().has_value();
+  if (hasDisplaySizeValue) {
+    JS::NativeRNCImageEditor::SpecCropImageCropDataDisplaySize rawDisplaySize = *data.displaySize(); // Extract the value from the optional
     // in pixels
-    targetSize = [RCTConvert CGSize:@{ @"width": @(displaySize.width()), @"height": @(displaySize.height()) }];
+    displaySize = [RCTConvert CGSize:@{ @"width": @(rawDisplaySize.width()), @"height": @(rawDisplaySize.height()) }];
   }
-  NSString *displaySize = data.resizeMode();
+  RCTResizeMode resizeMode = [RCTConvert RCTResizeMode:data.resizeMode() ?: @"contain"];
```

Comment:
That logic is still duplicated. How about introducing some "adapter" function that takes arguments on both archs and returns some common struct/object that keeps all the arguments with desired ObjC types?

I drafted sth like this, it compiles and seems to be working. Might be a good starting point:

```objectivec
// constant for default quality value from another PR
#define DEFAULT_COMPRESSION_QUALITY 0.9

// I used C++ struct for simplicity, but ObjC NSObject subclass will be also be ok
struct Args {
public:
    CGPoint offset;
    CGSize size;
    CGSize displaySize;
    RCTResizeMode resizeMode;
    double quality;
    NSString *format;
};

// ...

@implementation RNCImageEditor

// Here I declare helper "adapter" function that returns my struct
// the trick is to make all arguments an (id) type,
// which in ObjC is an equivalent of TS any type
- (Args)adaptArgsWithFormat:(id)format
                      width:(id)width
                     height:(id)height
                    offsetX:(id)offsetX
                    offsetY:(id)offsetY
                 resizeMode:(id)resizeMode
               displayWidth:(id)displayWidth
              displayHeight:(id)displayHeight
                    quality:(id)quality
{
    // And here I just return an object with args as properties
    // the trick is to use RCTConvert helper to convert (id) to proper ObjC type
    return Args{
        .offset = {[RCTConvert double:offsetX], [RCTConvert double:offsetY]},
        .size = {[RCTConvert double:width], [RCTConvert double:height]},
        .displaySize = {[RCTConvert double:displayWidth], [RCTConvert double:displayHeight]},
        .resizeMode = [RCTConvert RCTResizeMode:resizeMode ?: @"contain"],
        .quality = [RCTConvert CGFloat:quality],
        .format = [RCTConvert NSString:format]
    };
}

// ...

// Usage in new arch
- (void) cropImage:(NSString *)uri
         cropData:(JS::NativeRNCImageEditor::SpecCropImageCropData &)data
         resolve:(RCTPromiseResolveBlock)resolve
         reject:(RCTPromiseRejectBlock)reject
{
    // All primitive types like double or float need to wrapped in @()
    auto args = [self adaptArgsWithFormat:data.format()
        width:@(data.size().width())
        height:@(data.size().height())
        offsetX:@(data.offset().x())
        offsetY:@(data.offset().y())
        resizeMode:data.resizeMode()
        displayWidth:@(data.displaySize().has_value() ? data.displaySize()->width() : 0) 
        displayHeight:@(data.displaySize().has_value() ? data.displaySize()->height() : 0) 
        quality:@(data.quality().has_value() ? *data.quality() : DEFAULT_COMPRESSION_QUALITY)];

// ...

// Usage in old arch
RCT_EXPORT_METHOD(cropImage:(NSURLRequest *)imageRequest
                  cropData:(NSDictionary *)cropData
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)
{
    auto args = [self adaptArgsWithFormat:cropData[@"format"]
        width:cropData[@"size"][@"width"]
        height:cropData[@"size"][@"height"]
        offsetX:cropData[@"offset"][@"x"]
        offsetY:cropData[@"offset"][@"y"]
        resizeMode:cropData[@"resizeMode"]
        displayWidth:cropData[@"displaySize"] ? cropData[@"displaySize"][@"x"] : @(0)
        displayHeight:cropData[@"displaySize"] ? cropData[@"displaySize"][@"y"] : @(0)
        quality:cropData[@"quality"] ? cropData[@"quality"] : @(DEFAULT_COMPRESSION_QUALITY)];

// ...

// Reading data from struct

CGRect rect = {args.offset, args.size};

```

Reply:
as I see, codegen already create typed structure and 
we don't need and extra data transformation to `struct Args`  
when an old arch support will be removed


can we keep an existing code as there not planned much changes args struct?

Reply:
The old arch compatibility will need to be kept for a long time (maybe even a year or two) so to prevent future bugs (like this one) and to prevent duplication of code, we should have such kind of transformation helper (as the codegen structs cannot be used for old arch)

Reply:
@mateusz1913 ok, can you review again